### PR TITLE
Document hash partition count property

### LINKED ISSUE #11256: Document hash partition property
See https://github.com/trinodb/trino/pull/11204 for info.

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Minimal documentation for the property and session property.

> Is this change a fix, improvement, new feature, refactoring, or other?

improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

core settings

> How would you describe this change to a non-technical end user or system administrator?

Add minimal documentation for setting the partition count used for distributed query processing. 

## Related issues, pull requests, and links

* Fixes #11256
* Related to #11204

## Documentation

(✅ ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅ ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 1]
```diff
@@ -12,16 +12,27 @@ Query management properties
...
+
+``query.hash-partition-count``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Default value:** ``100``
+* **Session property:** ``hash_partition_count``
+
+The number of partitions to use for processing distributed joins and
+aggregations.
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-query-management.rst:34]: Do we want to offer advice on what higher and lower numbers mean .. e.g. make the number to low and query processing will not be able to parallelize enough. Make the number too high and ?? .. also how much should it correspond with number of workers (I think not at all but...)

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 2]
```diff
@@ -12,16 +12,27 @@ Query management properties
 Configures the algorithm to organize the processing of all of the
 stages of a query. You can use the following execution policies:
 
-* ``phased`` schedules stages in a sequence to avoid blockages because of 
-  inter-stage dependencies. This policy maximizes cluster resource utilization 
+* ``phased`` schedules stages in a sequence to avoid blockages because of
```
Comment (sopel39) [inline docs/src/main/sphinx/admin/properties-query-management.rst:15]: extract formatting changes to separate commit

File: docs/src/main/sphinx/admin/properties-query-management.rst (hunk 1 above)
Comment (sopel39) [reply docs/src/main/sphinx/admin/properties-query-management.rst:34]: I wouldn't offer advice yet since we don't have any data on that yet. Ideally it should be adaptive or driven by CBO

**Review by sopel39 — APPROVED** [review]
lgtm % comment

File: docs/src/main/sphinx/admin/properties-query-management.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-query-management.rst:34]: ok

File: docs/src/main/sphinx/admin/properties-query-management.rst (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-query-management.rst:15]: Turns out neither IDEA nor Visual Code let me save the file with the trailing spaces anymore.

Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-query-management.rst:15]: Had to resort to vim .. haha.. done now.

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 3]
```diff
@@ -22,6 +22,16 @@ stages of a query. You can use the following execution policies:
...
   the query wall time as it attempts to minimize the number of running stages.
 
+``query.hash-partition-count``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Default value:** ``100``
+* **Session property:** ``hash_partition_count``
+
+The number of partitions to use for processing distributed joins and
```
Comment (sopel39) [inline docs/src/main/sphinx/admin/properties-query-management.rst:32]: more operators are involved (e.g. window function). I would make this sentence a bit more generic, e.g. `to use for processing distributed operations (e.g. join, aggregation, partitioned window function).`

Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-query-management.rst]: Done ..

**Comment by mosabua** [comment]
@sopel39 @findepi @martint can we get this merged so it gets into the same release as the code change..
